sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -20,247 +19,399 @@
-}
+
+} // namespace sycl::ext::oneapi::experimental::matrix
 ```
-The matrix multiply and add function performs the multiply operation on the matrices `A` and `B`, accumulate the result with `C` and return the result.
+The matrix multiply and add function performs the multiply operation
+on the matrices `A` and `B`, accumulates the result with `C` and returns
+the result.
 
 
-#### Matrix Initialization: `joint_matrix_fill`
-The current interface presented above assumes that all the matrices are directly loaded from memory. This new function called `joint_matrix_fill`  makes it possible to multiply a matrix which is not directly loaded from memory but rather initialized directly in the register. On Intel AMX, if the initialization constant is zero, this would map to the `_tile_zero` intrinsic: 
+==== Matrix Initialization: `joint_matrix_fill`
+Unlike `joint_matrix_load` that assumes that all the matrices are
+directly loaded from memory, `joint_matrix_fill`  makes it possible to
+multiply a matrix which is not directly loaded from memory but rather
+initialized directly in the register. On Intel AMX, if the
+initialization constant is zero, this would map to the `_tile_zero`
+intrinsic. Note that the value type `Tv` must be convertible to the
+matrix elements type `T`.
 
 ```c++
 namespace sycl::ext::oneapi::experimental::matrix {
-  template <typename Group, typename T, size_t NumRows, size_t NumCols,
-           use Use, layout Layout, typename Tv>
-  void joint_matrix_fill(Group sg, joint_matrix<Group, T, Use, NumRows, NumCols, Layout> &m, Tv v);
-}
-```
-IMPORTANT: In the current implementation, only the `sub_group` scope is supported.  
 
-#### Element Indexing and Piece-Wise Operations
-##### Background
-Besides matrix multiply and add, this extension aims to make it possible to perform piece-wise operations on matrices in a SPMD manner. The mechanisms that are recommended to perform such piece-wise operations depend upon which of the following classes the operation falls into:
+template <typename Group, typename T, size_t Rows, size_t Cols,
+          use Use, layout Layout, typename Tv>
+  void joint_matrix_fill(Group g, joint_matrix<Group, T, Use,
+                         Rows, Cols, Layout> &m, Tv v);
 
-Class 1- Element-wise operations where the same operation is performed on every element of the matrix, such that the operation can be performed without knowledge of the position of the element within the matrix. Activation functions or adding a constant value to every element of the matrix are two examples.
+} // namespace sycl::ext::oneapi::experimental::matrix
+```
 
-Class 2- Piece-wise operations where the operation depends on the element index of the matrix or the operation takes multiple elements as operands (such as a sum of all elements in a row for example). Quantization that is needed for conversion between low precision types like `int8_t` and `fp32` uses piece-wise operations.
+==== Element-Wise Operations
```

Comment:
what do you mean element-wise operations is a "slang" phrase? this is the term used in BLAS library specifications

Reply:
It seems like slang English to me, but I don't feel that strongly especially if it's used in other literature.  Let's be consistent, though.  The sycl_ext_intel_matrix spec use "per-element", so either change both to use "element-wise" or both to use "per-element".

Reply:
I have "per-element" in the sycl_ext_intel_matrix spec because the variant with the coordinates let you do per element access and operation.
However, the plain `joint_matrix_apply` can only do element-wise operations. You cannot really access one specific element and make special operations. 
I think it is better to make this distinction. Otherwise, why we would have a new functionality in the Intel document if it does not provide more.

Reply:
I think you are saying that the term "per-element" means that the API provides coordinates while the term "element-wise" means that the API does not provide coordinates.

In my mind, the two terms are equivalent, and there is no such distinction.  I think many other readers will be the same.  If you want to make this distinction, you should be more explicit like "Per-element access with coordinates".